handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1557,14 +1559,26 @@ public final synchronized void setSSLParameters(SSLParameters sslParameters) {
 
         int version = PlatformDependent.javaVersion();
         if (version >= 7) {
-            endPointIdentificationAlgorithm = sslParameters.getEndpointIdentificationAlgorithm();
+            final String endPointIdentificationAlgorithm = sslParameters.getEndpointIdentificationAlgorithm();
+            final boolean endPointVerificationEnabled = endPointIdentificationAlgorithm != null &&
+                    !endPointIdentificationAlgorithm.isEmpty();
+            SSL.setHostNameValidation(ssl, DEFAULT_HOSTNAME_VERIFICATION_FLAGS,
+                    endPointVerificationEnabled ? getPeerHost() : null);
```

Comment:
https://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLParameters.html#setEndpointIdentificationAlgorithm-java.lang.String-
> If the algorithm parameter is non-null or non-empty, the endpoint identification/verification procedures must be handled during SSL/TLS handshaking.

OpenSSL Engine support HTTPS and LDAPS as defined in https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html. As you can see I added support for defines in tcnative, but I'm not yet sure we need to use them.